signac/__main__.py

```diff
@@ -1042,20 +1042,16 @@ def jobs():
             if READLINE:
                 if 'PyPy' not in platform.python_implementation():
                     fn_hist = project.fn('.signac_shell_history')
-                    write_access = os.access(fn_hist, os.W_OK)
-                    read_access  = os.access(fn_hist, os.R_OK)
-                    if not os.path.isfile (fn_hist):
-                        atexit.register(readline.write_history_file, fn_hist)
-                    elif not write_access and not read_access:
-                        print ("Warning: .signac_shell_history does not have read and write permission. The history of this shell will not be saved.")
-                    elif not write_access:
-                        print ("Warning: .signac_shell_history does not have write premission. The history of this shell will not be saved.")
-                    elif not read_access:
-                        print ("Warning: .signac_shell_history does not have read premission. The history of this shell will not be saved.")
-                    else:
+                    try:
                         readline.read_history_file(fn_hist)
                         readline.set_history_length(1000)
+                        with open(fn_hist, 'w') as f: pass
+                        atexit.register(readline.write_history_file, fn_hist)
+                    except FileNotFoundError:
                         atexit.register(readline.write_history_file, fn_hist)
+                    except PermissionError:
+                        print("Warning: .signac_shell_history does not have read/write permission.\
```

Comment:
Try breaking the line like this. Python will combine the two strings into one message. Notice there is a space at the end of the first line.
```python
print("Warning: .signac_shell_history does not have read/write permission. "
      "The history of this shell will not be saved.")

Reply:
@bdice done!